Solution1/DataAccessLayer/Repository/WetherRepository.cs

```diff
@@ -0,0 +1,30 @@
+﻿using DataAccessLayer.Abstract;
+using DataAccessLayer.Model;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DataAccessLayer.Repository
+{
+    public class WetherRepositor : IWetherRepository
+    {
+        public Weather GetByCityName(string cityName)
+        {
+            var request = WebRequest.Create($"https://api.openweathermap.org/data/2.5/weather?q={cityName}&appid={Const.KeyApi}&units=metric");
```

Comment:
not in DataAccessLayer - weather is not data of your application, this is data of openweathermap, it can't be here